tests/dhcp_server/test_dhcp_server.py

```diff
@@ -0,0 +1,336 @@
+import binascii
+import logging
+import pytest
+import ptf.packet as scapy
+import ptf.testutils as testutils
+from tests.common.utilities import capture_and_check_packet_on_dut
+from tests.common.helpers.assertions import pytest_assert
+
+
+pytestmark = [
+    pytest.mark.topology('m0', 'mx'),
+]
+
+
+VLAN_GATE_WAY = '192.168.0.1'
+VLAN_SUBNET_MASK = '255.255.255.0'
+DHCP_DEFAULT_LEASE_TIME = 300
+DHCP_DEFAULT_CUSTOM_OPTION_VALUE = 'hello_from_sonic'
+DHCP_MAC_BROADCAST = "ff:ff:ff:ff:ff:ff"
+DHCP_IP_DEFAULT_ROUTE = "0.0.0.0"
+DHCP_IP_BROADCAST = "255.255.255.255"
+DHCP_UDP_CLIENT_PORT = 68
+DHCP_UDP_SERVER_PORT = 67
+DHCP_MESSAGE_TYPE_DISCOVER_NUM = 1
+DHCP_MESSAGE_TYPE_OFFER_NUM = 2
+DHCP_MESSAGE_TYPE_REQUEST_NUM = 3
+DHCP_MESSAGE_TYPE_ACK_NUM = 5
+DHCP_MESSAGE_TYPE_NAK_NUM = 6
+
+
+def clean_fdb_table(duthost):
+    duthost.shell("sonic-clear fdb all")
+
+
+def ping_dut_refresh_fdb(ptfhost, interface, vlan_ip):
+    ptfhost.shell("timeout 1 ping -c 1 -w 1 -I {} {}".format(interface, vlan_ip), module_ignore_errors=True)
+
+
+def clean_dhcp_server_config(duthost):
+    keys = duthost.shell("sonic-db-cli CONFIG_DB KEYS DHCP_SERVER_IPV4*")
+    for key in keys["stdout_lines"]:
+        duthost.shell("sonic-db-cli CONFIG_DB DEL '{}'".format(key))
+
+
+@pytest.fixture(scope="module", autouse=True)
+def dhcp_server_config_setup_and_teardown(duthost):
+    clean_dhcp_server_config(duthost)
+    config_commands = [
+                        'config dhcp_server ipv4 add --mode PORT --lease_time %s ' % DHCP_DEFAULT_LEASE_TIME +
+                        '--gateway %s --netmask 255.255.255.0 Vlan1000' % VLAN_GATE_WAY,
+                        'config dhcp_server ipv4 enable Vlan1000',
+                        # config part 1: for single ip assignment test
+                        'config dhcp_server ipv4 range add test_single_0 192.168.0.10',
```

Comment:
Better to config ip assigned based on vlan ip which is got from config. Suggest using python lib `ipaddress` to caculate

Reply:
Yes, added code to parse VLAN and port relationship between DUT and PTF from running config instead of const.